### PR TITLE
[RFR] [BC Break] Rename admin-on-rest to react-admin

### PR DESCRIPTION
**Pull request by fzaninotto**
- [x] [BC Break] rename admin-on-rest to react-admin
- [x] [BC Break] rename `restClient` to `dataProvider`
- [x] Split code into packages (using Yarn workspaces)
- [x] [BC Break] Move `simpleRestClient` and `jsonServerRestClient` to standalone packages
- [x] [BC Break] Change the key of core translation messages from "aor" to "ra"
- [x] [BC Break] Move English messages to a standalone package
- [x] [BC Break] Remove `addField` prop (replace by `addField` HOC)
- [x] [BC Break] Rename CSS classes (`aor-field` => `ra-field`, `aor-input` => `ra-input`)
- [x] Make example work
- [x] Replace `mocha` by `jest`
- [x] Make unit tests pass
- [x] Make e2e tests pass
- [x] Update documentation
- [x] Write upgrade guide

### CODE REVIEW

File: UPGRADE.md [hunk 1]
```diff
@@ -1,5 +1,25 @@
 # Upgrade to 2.0
 
+## Admin-on-rest Renamed to React-Admin
+
+The main package name has changed, so you must update your dependencies:
+
+```sh
+npm uninstall -S admin-on-rest
+npm install -S react-admin
```
Comment (djhi) [inline UPGRADE.md]: `-S` is not needed anymore. npm now save by default like yarn

**Comment by oliviertassinari** [comment]
🎉

**Comment by fzaninotto** [comment]
Switching to RFR

File: UPGRADE.md [hunk 2]
```diff
@@ -1,5 +1,118 @@
...
+
+// before
+import { jsonServerRestClient } from 'admin-on-rest';
+// after
+import jsonServerRestClient from 'ra-data-json-server';
+```
+
+## Default (English) Messages Moved To Standalone Package
+
+The English messages have moved to another package, `ra-language-english`. The core package still displays the interface messages in English by default (by using `ra-language-english` as a dependency), but if you overrode some of the messages, you'll need to update the package name:
```
Comment (djhi) [inline UPGRADE.md]: s/overrode/override/

File: UPGRADE.md [hunk 3]
```diff
@@ -1,5 +1,118 @@
...
+const messages = { 'en': enMessages };
+
+<Admin locale="en" messages={messages}>
+  ...
+</Admin>
+```
+
+## Message Hash Main Key Changed ("aor" => "ra")
+
+The main kety of translation message objects was renamed from "aor" ro "ra". You must update your custom messages accordingly if you overrode core interface messages. If you're a language package author, you must also update and  republish your package to have it work with react-admin 2.0.
```
Comment (djhi) [inline UPGRADE.md]: s/kety/key/

File: UPGRADE.md [hunk 4]
```diff
@@ -1,5 +1,118 @@
...
+const messages = { 'en': enMessages };
+
+<Admin locale="en" messages={messages}>
+  ...
+</Admin>
+```
+
+## Message Hash Main Key Changed ("aor" => "ra")
+
+The main key of translation message objects was renamed from "aor" ro "ra". You must update your custom messages accordingly if you overrode core interface messages. If you're a language package author, you must also update and  republish your package to have it work with react-admin 2.0.
```
Comment (djhi) [inline UPGRADE.md]: s/overrode/override/

File: docs/Authorization.md [hunk 5]
```diff
@@ -5,9 +5,9 @@ title: "Authorization"
 
 # Authorization
 
-Some applications may require to determine what level of access a particular authenticated user should have to secured resources. Since there are many different possible strategies (single role, multiple roles or rights, etc.), admin-on-rest simply provides hooks to execute your own authorization code.
+Some applications may require to determine what level of access a particular authenticated user should have to secured resources. Since there are many different possible strategies (single role, multiple roles or rights, etc.), react-admin simply provides hooks to execute your own authorization code.
 
-By default, an admin-on-rest app doesn't require authorization. However, if needed, it will rely on the `authClient` introduced in the [Authentication](./Authentication.html) section.
+By default, an react-admin app doesn't require authorization. However, if needed, it will rely on the `authClient` introduced in the [Authentication](./Authentication.html) section.
```
Comment (djhi) [inline docs/Authorization.md]: s/an react-admin/a react-admin/

File: docs/DataProviders.md [hunk 6]
```diff
@@ -1,60 +1,60 @@
...
+* **[JSON server](https://github.com/typicode/json-server)**: [marmelab/ra-data-json-server](https://github.com/marmelab/ra-data-json-server) ([read more below](#json-server-rest)).
+* Local JSON: [marmelab/ra-data-fakerest](https://github.com/marmelab/ra-data-fakerest). Based on a local object, it doesn't even use HTTP. Use it for testing purposes.
 
-You can find REST clients for various backends in third-party repositories:
+You can find Data Providers for various backends in third-party repositories:
 
 * **[Epilogue](https://github.com/dchester/epilogue)**: [dunghuynh/aor-epilogue-client](https://github.com/dunghuynh/aor-epilogue-client)
 * **[Feathersjs](http://www.feathersjs.com/)**: [josx/aor-feathers-client](https://github.com/josx/aor-feathers-client)
 * **[Firebase](https://firebase.google.com/)**: [sidferreira/aor-firebase-client](https://github.com/sidferreira/aor-firebase-client)
 * **[GraphQL](http://graphql.org/)**: [marmelab/aor-simple-graphql-client](https://github.com/marmelab/aor-simple-graphql-client) (uses [Apollo](http://www.apollodata.com/))
```
Comment (djhi) [inline docs/DataProviders.md]: We should probably remove this one now

File: docs/Ecosystem.md [hunk 7]
```diff
@@ -21,16 +21,19 @@ title: "Ecosystem"
...
-- [kimkha/aor-loopback](https://github.com/kimkha/aor-loopback): a REST client works with [Loopback](http://loopback.io/)
-- [leperone/aor-parseserver-client](https://github.com/leperone/aor-parseserver-client): a REST client for [Parse Server](https://github.com/ParsePlatform/parse-server)
-- [moonlight-labs/aor-jsonapi-client](https://github.com/moonlight-labs/aor-jsonapi-client): a REST client for [JSON API](http://jsonapi.org/)
-- [mhdsyrwan/aor-rest-client-router](https://github.com/MhdSyrwan/aor-rest-client-router): a REST client that enables routing to many REST clients based on resource name.
+## Data Providers
+
+* **[Epilogue](https://github.com/dchester/epilogue)**: [dunghuynh/aor-epilogue-client](https://github.com/dunghuynh/aor-epilogue-client)
+* **[Feathersjs](http://www.feathersjs.com/)**: [josx/aor-feathers-client](https://github.com/josx/aor-feathers-client)
+* **[Firebase](https://firebase.google.com/)**: [sidferreira/aor-firebase-client](https://github.com/sidferreira/aor-firebase-client)
+* **[GraphQL](http://graphql.org/)**: [marmelab/aor-simple-graphql-client](https://github.com/marmelab/aor-simple-graphql-client) (uses [Apollo](http://www.apollodata.com/))
```
Comment (djhi) [inline docs/Ecosystem.md]: Deprecated!

File: docs/Translation.md [hunk 8]
```diff
@@ -63,24 +66,27 @@ You can find translation packages for the following languages:
 - Ukrainian (`uk`): [vitivs/aor-language-ukrainian](https://github.com/vitivs/aor-language-ukrainian)
 - Vietnamese (`vi`): [kimkha/aor-language-vietnamese](https://github.com/kimkha/aor-language-vietnamese)
 
-If you want to contribute a new translation, feel free to submit a pull request to update [this page](https://github.com/marmelab/admin-on-rest/blob/master/docs/Translation.md) with a link to your package.
+These packages are not directly interoperable with react-admin, but the upgrade is strightforward; rename the root key from "aor" to "ra". We invite the authors of the packages listed above to republish their translations for react-admin, using a different package name.
```
Comment (djhi) [inline docs/Translation.md]: s/strightforward/straightforward/

**Pull request merged**